core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1106 @@
+    ): QuadrantFractions | undefined => {
+      if (angle0 > angle1)
+        return convertQ1RadiansInSweepToQuadrantFractions(anglesInQ1, angle1, angle0, sweep);
+      if (Angle.isAlmostEqualRadiansNoPeriodShift(angle0, angle1))
+        return undefined; // empty sweep
+      const qData = QuadrantFractions.getQuadrantRadians(angle0, angle1);
+      if (undefined === qData)
+        return undefined; // no containing quadrant
+      const qFractions = new OrderedSet<number>(compareFractions);
+      const f0 = convertAndAddRadiansToFractionInRange(qFractions, angle0, sweep);
+      const f1 = convertAndAddRadiansToFractionInRange(qFractions, angle1, sweep);
+      if (undefined === f0 || undefined === f1)
+        return undefined;
+      for (const a0 of anglesInQ1) {
+        let angle = a0;
+        if (2 === qData.quadrant)
+          angle = Angle.piRadians - angle;
+        else if (3 === qData.quadrant)
+          angle = Angle.piRadians + angle;
+        else if (4 === qData.quadrant)
+          angle = Angle.pi2Radians - angle;
+        convertAndAddRadiansToFractionInRange(qFractions, angle, sweep, f0, f1);
+      }
+      const qf = QuadrantFractions.create(qData.quadrant, [...qFractions]);
+      const n = qf.fractions.length;
+      const qFrac0 = sweep.radiansToSignedPeriodicFraction(qData.angle0);
+      const qFrac1 = sweep.radiansToSignedPeriodicFraction(qData.angle1);
+      qf.axisAtStart = Geometry.isAlmostEqualEitherNumber(qf.fractions[0], qFrac0, qFrac1, Geometry.smallFraction);
+      qf.axisAtEnd = Geometry.isAlmostEqualEitherNumber(qf.fractions[n - 1], qFrac0, qFrac1, Geometry.smallFraction);
+      if (2 === n) {
+        qf.fractions.splice(1, 0, Geometry.interpolate(qf.fractions[0], 0.5, qf.fractions[1]));
+        qf.averageAdded = true;
+      }
+      return qf;
+    };
+    const computeStructuredOutput = (anglesInQ1: number[], arcSweep: AngleSweep): QuadrantFractions[] => {
+      const qEndAngles = new OrderedSet<number>(arcSweep.isCCW ? compareRadiansIncreasing : compareRadiansDecreasing);
+      qEndAngles.add(arcSweep.endRadians);
+      for (const qAngle of [0, Angle.piOver2Radians, Angle.piRadians, Angle.pi3Over2Radians, Angle.pi2Radians]) {
+        const shifted = shiftRadiansToSweep(qAngle, arcSweep);
+        if (shifted.inSweep)
+          qEndAngles.add(shifted.angle);
+      }
+      const quadrants = new OrderedSet<QuadrantFractions>(compareQuadrantFractions);
+      let a0 = arcSweep.startRadians;
+      for (const a1 of qEndAngles) {
+        const quadrant = convertQ1RadiansInSweepToQuadrantFractions(anglesInQ1, a0, a1, arcSweep);
+        if (quadrant)
+          quadrants.add(quadrant);
+        a0 = a1;
+      }
+      return [...quadrants];
+    };
+    const computeFlatOutput = (anglesInQ1: number[], arcSweep: AngleSweep): number[] => {
+      // Flat array output: first add quadrant fractions so the set prefers them over nearby interior fractions
```

Comment:
It's not super important but I try to be consistent with comments and start with lower case as you did for all above comments.

```suggestion
      // flat array output: first add quadrant fractions so the set prefers them over nearby interior fractions
```